Prevent a 2nd context from closing the 1st context’s log.

FGasper opened the pull request:
Issue #52

FGasper committed:
Prevent a 2nd context from closing the 1st context’s log.

Issue #52

wcawijngaards commented:
Your fix would stop all logfile closes.  That may work for you, but is not something for the generic code base.  The commit I just added creates a flag where it tracks if a debug file descriptor is in use, and when that is in use it tries to omit the file closures.  Also for the second context, the flag is a global variable.  I hope this fixes it.

wcawijngaards commented:
That did not work, but the second commit makes it work.  There was another log_init that needed to be stopped.  Does that also work for you, I mean, apart from the (very useful) example fail program?

FGasper commented:
@wcawijngaards My apologies for the non-responses here; with the time difference I think I just didn’t see the email notifications.

Thank you for looking into this and addressing the issues!
